services/backend/src/core/search/util/getAllProfiles.js

```diff
@@ -33,44 +33,49 @@ async function getAllUsers(searchValue, language, userId) {
           exFeeder,
         },
       }) => {
-        const Friends = await prisma.user.findOne({
+        const userConnections = await prisma.user.findOne({
           where: { id },
-          select: { friends: true },
+          select: { connections: true },
         });
 
-        isFriends = Friends.friends.some((item) => item.id === userId);
+        isConnection = userConnections.connections.some(
+          (item) => item.id === userId
+        );
 
         return {
           id,
           visibleCards: {
-            info: !(info === "PRIVATE" || (info === "FRIENDS" && !isFriends)),
+            info: !(
+              info === "PRIVATE" ||
+              (info === "CONNECTIONS" && !isConnection)
+            ),
```

Comment:
Something similar to the other comment, maybe export that logic into another function

```js
const visibleCardBool = (value) => !(value === "PRIVATE" || (value === "CONNECTIONS" && !isConnection));
```

Then the return would just be
```js
info: visibleCardBool(info),
```

If you want, you could approach it the same way as the other comment, with keys, you would just not decompose the values of `visibleCards` in the map.